models/issues/issue.go

```diff
@@ -2498,3 +2534,88 @@ func DeleteOrphanedIssues(ctx context.Context) error {
 func (issue *Issue) HasOriginalAuthor() bool {
 	return issue.OriginalAuthor != "" && issue.OriginalAuthorID != 0
 }
+
+// TimeEstimateFromStr returns time estimate in seconds from formatted string
+func (issue *Issue) TimeEstimateFromStr(timeStr string) int64 {
+	timeTotal := 0
+
+	// Time match regex
+	rOnlyHours := regexp.MustCompile(`^([\d]+)$`)
+	rWeeks := regexp.MustCompile(`([\d]+)w`)
+	rDays := regexp.MustCompile(`([\d]+)d`)
+	rHours := regexp.MustCompile(`([\d]+)h`)
+	rMinutes := regexp.MustCompile(`([\d]+)m`)
```

Comment:
These can be defined globally before the function definition, similar to 
https://github.com/go-gitea/gitea/blob/cbc9a0fe477b1b8af249ca0b8dac5fc2be64e9f6/modules/templates/helper.go#L787
```suggestion

var (
	// Time match regex
	rOnlyHours = regexp.MustCompile(`^([\d]+)$`)
	rWeeks = regexp.MustCompile(`([\d]+)w`)
	rDays = regexp.MustCompile(`([\d]+)d`)
	rHours = regexp.MustCompile(`([\d]+)h`)
	rMinutes = regexp.MustCompile(`([\d]+)m`)
)

// TimeEstimateFromStr returns time estimate in seconds from formatted string
func (issue *Issue) TimeEstimateFromStr(timeStr string) int64 {
	timeTotal := 0
```

Reply:
If these vars are used only once pre request and only in this function, why move them to the package scope?

Reply:
Because you recalculate them with every request at the moment.
With the package scope, you only calculate them once.

Reply:
> Because you recalculate them with every request at the moment. With the package scope, you only calculate them once.

Ah, alright. Thank you